src/loss.py

```diff
@@ -1,2 +1,32 @@
 from abc import ABC
+import numpy as np
 
+
+class Purity_function(ABC):
+    def __init__(self):
+        pass
+
+    def purity_value(self, label_values: np.array, mask) -> float:
+        pass
+
+
+class Information_gaing(Purity_function):
+    super().__init__()
+
+
+
+    def _entropy(self, label_values: np.array) -> float:
+        _, uniqueClassesCounts = np.unique(label_values, return_counts=True)
```

Comment:
Definitely use snake_case. It's preferred to unpack as:
```suggestion
        unique_classes, uniqueClassesCounts = np.unique(label_values, return_counts=True)
```
However in this case, since the unique_classes aren't used, it is correct to unpack it in a _

Reply:
That being said the namings could be better
```suggestion
        _, number_of_unique_elements_per_class = np.unique(label_values, return_counts=True)
```